lib/rack/body_proxy.rb

```diff
@@ -38,7 +43,18 @@ def closed?
 
     # Delegate missing methods to the wrapped body.
     def method_missing(method_name, *args, &block)
-      @body.__send__(method_name, *args, &block)
+      case method_name
+      when :to_str
+        super
+      when :to_ary
```

Comment:
For my own edification, why is it better to do it this way vs defining a method `to_ary` and delegating `respond_to?` to super?

Reply:
I wouldn't say either is necessarily better, though I prefer the approach of not overriding `respond_to?` in this case, as we are already overriding `respond_to_missing?` and `method_missing`.